pkg/proxy/apis/kubeproxyconfig/fuzzer/fuzzer.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fuzzer
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/google/gofuzz"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtimeserializer "k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/kubernetes/pkg/proxy/apis/kubeproxyconfig"
+	utilpointer "k8s.io/kubernetes/pkg/util/pointer"
+)
+
+// Funcs returns the fuzzer functions for the kube-proxy apis.
+func Funcs(codecs runtimeserializer.CodecFactory) []interface{} {
+	return []interface{}{
+		func(obj *kubeproxyconfig.KubeProxyConfiguration, c fuzz.Continue) {
+			c.FuzzNoCustom(obj)
+			obj.BindAddress = fmt.Sprintf("%d.%d.%d.%d", c.Intn(256), c.Intn(256), c.Intn(256), c.Intn(256))
+			obj.HealthzBindAddress = fmt.Sprintf("%d.%d.%d.%d:%d", c.Intn(256), c.Intn(256), c.Intn(256), c.Intn(256), c.Intn(65536))
+			obj.MetricsBindAddress = fmt.Sprintf("%d.%d.%d.%d:%d", c.Intn(256), c.Intn(256), c.Intn(256), c.Intn(256), c.Intn(65536))
+			obj.OOMScoreAdj = utilpointer.Int32Ptr(c.Int31())
+			obj.ResourceContainer = "/kube-proxy"
```

Comment:
Aren't most of these randomized already by `c.FuzzNoCustom(obj)`? Usually one only has to fixup some fields which have constraints that `c.FuzzNoCustom(obj)` does not know about, like maybe the addresses above.

Reply:
@sttts yeah I guess so. But I didn't identify which fields are already randomized. Which type(s) of the fields can't be done by `c.FuzzNoCustom(obj)`?

Reply:
> Which type(s) of the fields can't be done by c.FuzzNoCustom(obj)?

Read the source :) Generally it fuzzes quite a lot if not all.

Reply:
:+1: :smile:

Reply:
the two duration below can't be randomized?

Reply:
yes they can. (last time I tried to randomize them but failed. I believe I missed sth then)

Reply:
updated. Local bazel test passed.